Jenkinsfile

```diff
@@ -13,277 +13,35 @@
-              OsTools.runSafe(this, "docker image push ${organization}/${dockerHubRepository}:${version}")
-              OsTools.runSafe(this, "docker image push ${organization}/${dockerHubRepository}:latest")
-              OsTools.runSafe(this, "docker image push ${organization}/${dockerHubRepository}:${version}-slim")
-              OsTools.runSafe(this, "docker image push ${organization}/${dockerHubRepository}:latest-slim")
-            }
-
-            response = OsTools.runSafe(this, """
-            curl -X POST https://hub.docker.com/v2/users/login/ \
-              -H 'cache-control: no-cache' -H 'content-type: application/json' \
-              -d '{ "username": "${env.DOCKERHUB_API_USERNAME}", "password": "${env.DOCKERHUB_API_PASSWORD}" }'
-            """)
-            token = readJSON text: response
-            dockerHubApiToken = token.token
-
-            def readme = readFile file: 'README.md', encoding: 'UTF-8'
-            readme = readme.replaceAll("(?s)<!--.*?-->", "")
-            readme = readme.replace("\"", "\\\"")
-            readme = readme.replace("\n", "\\n")
-            readme = readme.replace("\\\$", "\\\\\$")
-            response = httpRequest customHeaders: [[name: 'authorization', value: "JWT ${dockerHubApiToken}"]],
-                acceptType: 'APPLICATION_JSON', contentType: 'APPLICATION_JSON', httpMode: 'PATCH',
-                requestBody: "{ \"full_description\": \"${readme}\" }",
-                url: "https://hub.docker.com/v2/repositories/${organization}/${dockerHubRepository}/"
-          }
-        }
-      }
-      stage('Push tags') {
-        withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: credentialsId,
-                          usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
-          OsTools.runSafe(this, "git tag ${version}")
-          OsTools.runSafe(this, """
-            git push \
-            https://${env.GITHUB_API_USERNAME}:${env.GITHUB_API_PASSWORD}@github.com/${organization}/${gitHubRepository}.git \
-              ${version}
-          """)
-        }
-        OsTools.runSafe(this, "git tag -d ${version}")
-      }
-    }
-  } finally {
-    OsTools.runSafe(this, "docker logout")
-    OsTools.runSafe(this, "docker system prune -a -f")
-    OsTools.runSafe(this, 'git clean -f && git reset --hard origin/master')
-  }
-}
-
-def readVersion() {
-  def content = readFile 'Dockerfile'
-  for (line in content.split('\n')) {
-    if (line.startsWith('ARG IQ_SERVER_VERSION=')) {
-      return getShortVersion(line.substring(22))
-    }
+@Library(['private-pipeline-library', 'jenkins-shared@test-expectations', 'iq-pipeline-library']) _
+
+import com.sonatype.jenkins.shared.Expectation
+
+dockerizedBuildPipeline(
+  lint: {
+    hadolint(['Dockerfile'])
+  },
+  buildAndTest: {
+    validateExpectations([
+          new Expectation('javaVersion', 'java', '-version', 'openjdk version "1.8.0_362"'),
```

Comment:
I might suggest a regex for this.  The version may vary slightly over time, which is ok.  For example, 

```groovy
new Expectation('java-version', 'java', '-version', /openjdk version \"11.0.*\" \d{4}-\d{2}-\d{2}/),
```